src/services/expenses.service.js

```diff
@@ -0,0 +1,97 @@
+'use strict';
+
+let expenses = [];
+const REQUIRE_KEYS = {
+  userId: 0,
+  spentAt: 'string',
+  title: 'string',
+  amount: 0,
+  category: 'string',
+  note: 'string',
+};
+
+const createExpense = (expenseData) => {
+  const expense = {
+    ...expenseData,
+    id: Number(new Date()),
+  };
+
+  expenses.push(expense);
+
+  return expense;
+};
+
+const getExpenses = () => {
+  return expenses;
+};
+
+const getExpenseById = (id) => {
+  return expenses.find(expense => expense.id === id);
+};
+
+const getExpenseByQuery = (query) => {
+  const {
+    userId,
+    categories,
+    from,
+    to,
+  } = query;
+
+  return expenses.filter(({
+    category,
+    userId: id,
+    spentAt,
+  }) => {
+    const spentDate = new Date(spentAt).valueOf();
+    const dateFrom = new Date(from).valueOf();
+    const dateTo = new Date(to).valueOf();
+
+    if (categories && category !== categories) {
+      return false;
+    }
```

Comment:
if there are no `categories` in search params we will get `undefined`
if only 1 `categories` we will get just value (....?categories=1)
And if we have many categories we will get array of values (....?categories=1&categories=2&categories=3)

![image](https://github.com/mate-academy/node_accounting-app-v2/assets/86789161/ea010564-92a2-4116-9fcf-a66739d6d01f)